### PR TITLE
merge develop into master

### PR DESCRIPTION
**Pull request by gregtalarico**

### CODE REVIEW

**Comment by benschiefer** [comment]
2 things
1. we should test with a large data set to make sure there are no adverse performance implications
2. we should ensure we handle sort/reorder column when a cell is selected

**Comment by gregtalarico** [comment]
@benschiefer did a bit of perf testing and did not notice any impact on performance. Also added the sort and reorder handling we talked about. please review

File: index.jsx [hunk 1]
```diff
@@ -269,6 +272,15 @@ var App = React.createClass({
         this.setState({})
     },
 
+    onCellClick: function(cell){
+        if (cell === null){
+            this.selectedCell = [];
+            return;
+        }
+
+        this.selectedCell = [cell];
```
Comment (benschiefer) [inline index.jsx]: selectedCell should be an individual item, should it not?  If it is expected to be an array, i would think the property should be named selectedCells

File: lib/Header/index.js [hunk 2]
```diff
@@ -380,6 +384,7 @@ module.exports = React.createClass({
         event.nativeEvent.stopSort = true;
 
         this.showFilterMenu(column, event);
+        // event.stopPropagation()
```
Comment (benschiefer) [inline lib/Header/index.js:387]: is this necessary/unnecessary?

File: lib/index.js [hunk 3]
```diff
@@ -259,6 +259,10 @@ module.exports = React.createClass({
     },
 
     onDropColumn: function onDropColumn(index, dropIndex) {
+        if (typeof this.props.onColumnOrderChange === 'function' && typeof this.props.onCellClick === 'function' && this.props.selectCells) {
+            this.props.onCellClick(null);
```
Comment (benschiefer) [inline lib/index.js]: shouldn't this be calling this.props.onSelectCells(null)?

File: src/index.jsx [hunk 4]
```diff
@@ -266,6 +266,10 @@ module.exports = React.createClass({
     },
 
     onDropColumn: function(index, dropIndex){
+        if (typeof this.props.onColumnOrderChange === 'function' && typeof this.props.onCellClick === 'function' && this.props.selectCells) {
+            this.props.onCellClick(null);
```
Comment (benschiefer) [inline src/index.jsx]: shouldn't this be callow this.props.onSelectCells

File: lib/Header/index.js (hunk 2 above)
Comment (gregtalarico) [inline lib/Header/index.js:387]: the build script does that. not sure why?

File: index.jsx (hunk 1 above)
Comment (gregtalarico) [inline index.jsx]: will udpate

File: lib/index.js (hunk 3 above)
Comment (gregtalarico) [inline lib/index.js]: will update

File: src/index.jsx (hunk 4 above)
Comment (gregtalarico) [inline src/index.jsx]: will update

**Comment by benschiefer** [comment]
looks :+1:

**Pull request merged**